project/sites/liofa/modules/custom/liofa_common/liofa_common.module

```diff
@@ -0,0 +1,48 @@
+<?php
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\node\NodeInterface;
+
+/**
+ * Implements hook_ENTITY_presave() for nodes.
```

Comment:
You can simplify this ever so slightly with this hook: hook_ENTITY_TYPE_presave()

It would mean you can skip the first check of whether $entity  implements NodeInterface because the hook is only triggered for that entity type.
 https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Entity%21entity.api.php/function/hook_ENTITY_TYPE_presave/10